src/Adapter/Order/CommandHandler/AddProductToOrderHandler.php

```diff
@@ -87,130 +96,142 @@ public function handle(AddProductToOrderCommand $command)
     {
         $order = $this->getOrderObject($command->getOrderId());
 
-        if ($this->context->cart !== null) {
-            $oldCartRules = $this->context->cart->getCartRules();
-        } else {
-            $oldCartRules = [];
-        }
+        $this->contextStateManager
+            ->setCurrency(new Currency($order->id_currency))
```

Comment:
It might be safer to override the customer as well Some operation rely on it, like getting the customer address which in turn allows to get the tax included/excluded parameter

Reply:
Done